### ISSUE TITLE
[Bug]: the parameter outline in page.pdf seems not work

### ISSUE THREAD

**Issue by zsq471214943** [puppeteer/puppeteer#12360]
### Minimal, reproducible example

```TypeScript
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch({
    executablePath:"D:\\tools\\.local-browser\\win64-1293398\\chrome-win\\chrome.exe",
    args:["--no-sandbox","--disable-setuid-sandbox","--generate-pdf-document-outline"],
    protocol:"cdp"
  });
  const page = await browser.newPage();
  await page.goto('file://D:\\temp\\filetest\\html\\测试pdf.html', {
    waitUntil: 'networkidle2',
  });
  // page.pdf() is currently supported only in headless mode.
  // @see https://bugs.chromium.org/p/chromium/issues/detail?id=753118
   await page.pdf({
    path: 'hn.pdf',
    format: 'letter',
    outline: true,
    timeout: 0
  });

  await browser.close();
})();
```

### Error string

no error

### Bug behavior

- [ ] Flaky
- [X] PDF

### Background

![image](https://github.com/puppeteer/puppeteer/assets/20215857/0e3e3dbe-86c2-46c4-875a-c21d44bc2f37)

### Expectation

the export pdf include the toc
![image_1](https://github.com/puppeteer/puppeteer/assets/20215857/b117868c-560c-4e50-a5cd-f0b13590c629)

### Reality

The result did not output as expected

### Puppeteer version

22.7.1

### Node version

20.12.2

### Package manager

npm

### Package manager version

10.5.0

### Operating system

Windows

**Comment by OrKoN** [comment]
outline parameter only works for the old headless mode (`headless: 'shell'`) Currently Chrome does not support it.

**Comment by OrKoN** [comment]
@Lightning00Blade could you please update the documentation to make sure the outline option remark properly show up on the website? Perhaps it makes sense to throw an error if it is used in the incompatible mode.

**Comment by OrKoN** [comment]
@Lightning00Blade let's also file a feature request to have this feature for Chrome and not only for the headless shell.

**Comment by Lightning00Blade** [comment]
There is already a bug on Chromium to add support for this option: https://issues.chromium.org/issues/41387522#comment48
Please upvote that feature request as it signals interest in the feature.

**Comment by OrKoN** [comment]
Looks like the feature is already implemented upstream and will likely ship with Chrome M126.

**Comment by OrKoN** [comment]
M126 is available with Puppeteer 22.11.0 and this feature is now supported (I will update the docs in another PR).
